doc/quickstart.rst

```diff
@@ -0,0 +1,187 @@
+       entry void done();
+     };
+     array [1D] Hello {
+       entry Hello();
+       entry void SayHi();
+     };            
+   };
+
+
+The hello.cpp file
+''''''''''''''''''
+
+The ``hello.cpp`` file contains the implementation of the mainchare and chare
+array declared in the ``hello.ci`` file above.
+
+.. code-block:: c++
+
+   #include "hello.decl.h" // created from hello.ci file above
+
+   /*readonly*/ CProxy_Main mainProxy;
+   constexpr int nElements = 8;
+
+   /*mainchare*/
+   class Main : public CBase_Main
+   {
+   public:
+     Main(CkArgMsg* m)
+     {
+       //Start computation
+       CkPrintf("Running Hello on %d processors with %d elements\n", CkNumPes(), nElements);
+       CProxy_Hello arr = CProxy_Hello::ckNew(nElements);
+       mainProxy = thisProxy;
+       arr[0].SayHi(0);
+     };
+
+     void done()
+     {
+       // Finish computation
+       CkPrintf("All done.\n");
+       CkExit();
+     };
+   };
+
+   /*array [1D]*/
+   class Hello : public CBase_Hello 
+   {
+   public:
+     Hello() {}
+
+     void SayHi()
+     {
+       CkPrintf("PE %d says: Hello world from element %d.\n", CkMyPe(), thisIndex);
+       if (thisIndex < nElements-1) {
+         thisProxy[thisIndex+1].SayHi(); // Pass the hello on
+       } else {
+         mainProxy.done(); // We've been around once -- we're done.
+       }
+     }
+   };
+
+   #include "hello.def.h" // created from hello.ci file above
+
+
+Compiling the example
+'''''''''''''''''''''
+
+Charm++ has a compiler wrapper, ``charmc``, to compile Charm++ applications. Please see
+Section :numref:`sec:compile` for more information about ``charmc``.
+
+.. code-block:: console
+
+   $ charm/bin/charmc hello.ci # creates hello.def.h and hello.decl.h
+   $ charm/bin/charmc hello.cpp -o hello
+
+
+Running the example
+'''''''''''''''''''
+
+Charm++ applications are started via ``charmrun``,
+which is automatically created by the ``charmc`` command above.
+Please see Section :numref:`sec:run` for more information about ``charmrun``.
+
+To run the application on two processors, use the following command:
+
+.. code-block:: console
+
+   $ ./charmrun +p2 ./hello
+   Charmrun> scalable start enabled.
+   Charmrun> started all node programs in 1.996 seconds.
+   Charm++> Running in non-SMP mode: 1 processes (PEs)
+   Converse/Charm++ Commit ID: v6.9.0-172-gd31997cce
+   Charm++> scheduler running in netpoll mode.
+   CharmLB> Load balancer assumes all CPUs are same.
+   Charm++> Running on 1 hosts (1 sockets x 4 cores x 2 PUs = 8-way SMP)
+   Charm++> cpu topology info is gathered in 0.000 seconds.
+   Running Hello on 2 processors with 8 elements
+   PE 0 says Hi from element 0
+   PE 0 says Hi from element 1
+   PE 0 says Hi from element 2
+   PE 0 says Hi from element 3
+   PE 1 says Hi from element 4
+   PE 1 says Hi from element 5
+   PE 1 says Hi from element 6
+   PE 1 says Hi from element 7
+   All done
+   [Partition 0][Node 0] End of program
+
+
+Where to go from here
```

Comment:
This should be title cased for consistency.

Reply:
Done